### ISSUE TITLE
[Bug]: elastic_transform2d is not identity for zero displacement with align_corners=False

### ISSUE THREAD

**Issue by AyushSrivastava1818** [kornia/kornia#4235]
## What happened?

`elastic_transform2d` does not preserve the input image when the displacement field is zero and `align_corners=False` is used.

A zero displacement field represents an identity deformation, so the output should be equal to the input up to floating-point tolerance.

The current implementation constructs the sampling grid using `create_meshgrid(h, w)` and then passes it to `torch.nn.functional.grid_sample` with the selected `align_corners` value.

Since `create_meshgrid` produces the corner-aligned normalized `[-1, 1]` grid, this grid is not consistent with the coordinate convention used by `grid_sample` when `align_corners=False`.

As a result, even with an exactly zero displacement field, the image is resampled at shifted coordinates.

### Expected behavior

With:

python
noise = torch.zeros(B, 2, H, W)

the transformation should be an identity transformation:
elastic_transform2d(image, noise, ...) == image
within numerical tolerance.

Actual behavior

On the current main checkout, using a 4x4 deterministic input and a zero displacement field produces a maximum absolute difference of:
tensor(11.2500)
So the input is substantially changed even though the displacement is exactly zero.

This behavior is specific to the align_corners=False coordinate convention; the same corner-aligned grid is consistent with align_corners=True.

### Minimal reproduction

```python
import torch
from kornia.geometry.transform import elastic_transform2d

image = torch.arange(16.0).reshape(1, 1, 4, 4)
noise = torch.zeros(1, 2, 4, 4)

output = elastic_transform2d(
    image,
    noise,
    kernel_size=(3, 3),
    sigma=(1.0, 1.0),
    alpha=(1.0, 1.0),
)

print("Input:")
print(image)

print("\nOutput:")
print(output)

print("\nMax absolute difference:")
print((output - image).abs().max())
```

### Environment

Kornia 0.9.0rc1 | PyTorch 2.14.0+cpu | Python 3.13.11 | Windows

**Comment by aminmiral** [comment]
Picking this up — @AyushSrivastava1818 say the word if you're already mid-way and I'll drop it.

Reproduced on `main` (`8f292308`), torch 2.14.0, CPU:

```
align_corners=False  max|out-in| = 11.25
align_corners=True   max|out-in| = 0.0
```

so it is the `align_corners=False` path only, as reported.

**Comment by AyushSrivastava1818** [comment]
@aminmiral I am doing this you can pick any other!
